.github/workflows/deploy.yml

```diff
@@ -0,0 +1,149 @@
+      # https://github.community/t/reusable-workflow-with-strategy-matrix/205676/2.
+      machine-types:
+        required: false
+        type: string
+        default: "[\"ubuntu-latest\"]"
+        description: >
+            Stringified JSON array with the name of the type of machine(s) to run the workflow under, e.g.
+            "[\"ubuntu-latest\"]" or "[\"ubuntu-latest\", \"windows-latest\"]".
+      dotnet-version:
+        required: false
+        type: string
+        default: 6.0.*
+        description: Version of the .NET SDK to set up.
+      build-directory:
+        required: false
+        type: string
+        default: .
+        description: Path to the directory where a solution file can be found.
+      build-verbosity:
+        required: false
+        type: string
+        default: quiet
+        description: Verbosity parameter for dotnet build.
+      build-enable-code-analysis:
+        required: false
+        type: string
+        default: "true"
+        description: Indicates whether to enable static code analysis during dotnet build.
+      build-enable-nuget-caching:
+        required: false
+        type: string
+        default: "false"
+        description: >
+          If set to "true", the NuGet packages will be cached by GitHub. This is faster if you have a lot of NuGet
+          packages, but negligible or even slower if you only have a handful.
+      build-enable-npm-caching:
+        required: false
+        type: string
+        default: "false"
+        description: >
+          If set to "true", the NPM packages will be cached by GitHub. This is faster if you have a lot of NPM packages,
+          but negligible or even slower if you only have a handful.
+      build-cache-version:
+        required: false
+        type: string
+        default: "0"
+        description: >
+          Change this to any other value to alter the cache key, effectively invalidating the latest cache. This is the
+          easiest current way to force "clear" the cache
+          (https://github.community/t/how-to-clear-cache-in-github-actions/129038/5) until the associated issue
+          (https://github.com/actions/cache/issues/2) is resolved and the feature is available not just via the RESTful
+          API.
+      timeout-minutes:
+        required: false
+        type: number
+        default: 360
+        description: Configuration for the timeout-minutes parameter of the workflow. The 360 is GitHub's default.
+      app-name:
+        required: true
+        type: string
+        description: The name of the existing app service.
+      slot-name:
+        required: true
+        type: string
+        description: Name of the web app slot.
+      resource-group-name:
+        required: true
+        type: string
+        description: Name of the resource group.
+      url:
+        required: true
+        type: string
+        description: The URL of the web app slot.
+
+jobs:
```

Comment:
Use `pwsh` as the shell (since you didn't specify the shell, under Ubuntu the default bash will be used): https://lombiq.atlassian.net/wiki/spaces/DEV/pages/2545614855/Writing+GitHub+Actions+workflows+and+actions#:~:text=Prefer%20to%20use%20PowerShell%207%2B%20scripts. From looking at it, I don't think you'll need to change the scripts.